### ISSUE TITLE
[Rockon] Improve cleanup of failed multi-container Rockon installations

### ISSUE THREAD

**Issue by Hooverdan96** [rockstor/rockstor-core#2929]
During testing of a multi-container Rockon PR ([#393](https://github.com/rockstor/rockon-registry/pull/393), the last of three containers was failing during the pull stage. The WebUI correctly reports that the installation failed and advises checking the logs.

However, the first two containers (in this example, based on a redis and a postgresql image, respectively) continued running, as they were successfully pulled and started.

Expected behavior would be that, if any of the containers defined in a Rockon failed during the installation, any already installed and running containers should be stopped and removed (I would say, also the images should be cleaned up, but that could be optional, though having unused images sticking around could eventually have an impact on storage space in the Rockon root).

This should probably be part of a larger effort addressing the Rockon management after the next major release. In the likely absence of that for some time, it might be useful to add some commentary to the Rockon documentation, adding the need/requirement to manually clean up dangling docker containers/images as to not impact system CPU/memory/storage capacity negatively.

**Comment by phillxnet** [comment]
@Hooverdan96 Yes, good points re multi-container fragility. Had similar instances during that draft BareOS work. The following was a good find now we are on newer docker, see: https://github.com/rockstor/rockon-registry/pull/383#issuecomment-2331364653


"During development we can end-up with some kipple:"

> docker system prune -a --volumes

This at least can help with removing old stuff that tends to build up. I.e. in your context of:

> ... adding the need/requirement to manually clean up dangling docker containers/images as to not impact system CPU/memory/storage capacity negatively.
